src/core_plugins/kibana/public/home/index.js

```diff
@@ -49,6 +51,7 @@ function getRoute() {
         const getter = indexPatterns.getIds;
         getter.clearCache();
       };
+      $scope.bulkCreate = savedObjectsClient.bulkCreate;
```

Comment:
I think we made a mistake by coupling rest clients to Angular in the past. Instead they should have been framework-agnostic singletons.

If we keep coupling new code (in this case React) to Angular we make it much harder to switch away from Angular, and increase technical debt.

I can see how it's easier to just use an existing client but I don't think the alternative is much more difficult. You don't have to re-implement there entire savedObjectsClient - just create a starting point with the endpoints you need.

Afaict this is all you need:
```
export function bulkCreate(savedObjects, { overwrite }) {
  return kfetch({
    pathname: `/api/saved_objects/_bulk_create`,
    method: 'POST',
    query: {
      overwrite
    },
    body: JSON.stringify(savedObjects)
  });
}
```

Reply:
@epixa Do we already have a strategy for consuming Angular services from outside Angular - and how to migrate them? I've already spent a fair amount of time de-angularizing some services and it would be beneficial if everyone were aligned. If we don't have anything I'm open for doing a write-up of the approach I've used, and how I see us moving forward.

Reply:
I do not think this is a problem and is exactly how we are interacting with saved objects in other react components. From react's perspective, it is just an injected function. Eventually, the savedObjectClient will be decoupled from angular but all saved object interaction should go through the savedObjectClient.

Reply:
The savedObjectClient converts the results into savedObject classes and camel cases the object keys. But more importantly, it puts all calls to the saved object API in a single place so if the API ever changes, only the savedObjectClient has to be updated instead of hunting down lots of segregated fetches.

Reply:
>  But more importantly, it puts all calls to the saved object API in a single place so if the API ever changes, only the savedObjectClient has to be updated instead of hunting down lots of segregated fetches

Totally agree. IMHO it is okay temporarily to have two "clients": the old Angular client, and a new framework agnostic service. Over time the Angular client could even call the framework agnostic client, and eventually the Angular client would be dropped. That's just my view, but interested to hear what other ideas are floating around.

Reply:
Btw. I think it's fine to leave it as is. Just my opinion on how to do it going forward.